lib/cosmos/utilities/sleeper.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: ascii-8bit
+
+# Copyright 2014 Ball Aerospace & Technologies Corp.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+
+module Cosmos
+
+  class Sleeper
```

Comment:
Sleeper implements a breakable sleep.  It is implemented using what is called the self-pipe trick.  This allows classes to sleep(1) but, have the sleep be interruptable so that they can be shutdown cleanly and quickly.